buildSrc/src/main/java/org/opensearch/gradle/testclusters/OpenSearchCluster.java

```diff
@@ -387,20 +392,27 @@ public void goToNextVersion() {
         writeUnicastHostsFiles();
     }
 
+    public void upgradeAllNodesAndPluginsToNextVersion(List<Provider<RegularFile>> plugins) {
+        stop(false);
+        nodes.all(OpenSearchNode::goToNextVersion);
+        upgradePlugin(plugins);
+        start();
+        writeUnicastHostsFiles();
```

Comment:
Curious: Do we need to `writeUnicastHostFiles`? 
Unicast is used for discovery of cluster nodes, if the cluster has the same ports already published and written to `unicast_host.txt` and upgrading the nodes the new version, wouldnt the framework start the new version of OpenSearch with the same ports?

Reply:
> This is awesome, plugins would love it and great work!
> Thanks a lot @VachaShah !!

Thank you @saratvemulapalli!